extensions/mssql/src/objectManagement/commands.ts

```
@@ -237,6 +241,35 @@ async function handleRenameObjectCommand(context: azdata.ObjectExplorerContext,
 	});
 }
 
+async function handleDetachDatabase(context: azdata.ObjectExplorerContext, service: IObjectManagementService): Promise<void> {
+	const connectionUri = await getConnectionUri(context);
+	if (!connectionUri) {
```

we should surface the error, otherwise the user would think it is not doing anything or hanging.

---

Looks like we already show an error notification as part of getConnectionUri, but we don't throw an Error.

---

you are right, this is good enough.